NetCord/Gateway/GatewayClient.cs

```diff
@@ -17,71 +20,385 @@ public partial class GatewayClient : WebSocketClient, IEntity
     public event Func<GuildIntegrationDeleteEventArgs, ValueTask>? GuildIntegrationDelete;
+
+    /// <summary>
+    /// Sent when a new invite to a channel is created. Only sent if the bot has the <see cref="Permissions.ManageChannels"/> permission for the relevant channel.
+    /// </summary>
     public event Func<GuildInvite, ValueTask>? GuildInviteCreate;
+
+    /// <summary>
+    /// Sent when an invite is deleted. Only sent if the bot has the <see cref="Permissions.ManageChannels"/> permission for the relevant channel.
+    /// </summary>
     public event Func<GuildInviteDeleteEventArgs, ValueTask>? GuildInviteDelete;
+
+    /// <summary>
+    /// Sent when a message is created. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// The inner payload is a message object with set <see cref="Message.GuildId"/>, <c>member</c> and <c>mentions</c> fields.
+    /// </summary>
     public event Func<Message, ValueTask>? MessageCreate;
+
+    /// <summary>
+    /// Sent when a message is updated. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// The inner payload is a message object with the same extra fields as <see cref="MessageCreate"/>. Updates may contain only a partial <see cref="Message"/> object (but will always contain an <c>ID</c> and <c>channel_id</c>).
+    /// </summary>
     public event Func<Message, ValueTask>? MessageUpdate;
+
+    /// <summary>
+    /// Sent when a message is deleted. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// </summary>
     public event Func<MessageDeleteEventArgs, ValueTask>? MessageDelete;
+
+    /// <summary>
+    /// Sent when multiple messages are deleted at once.
+    /// </summary>
     public event Func<MessageDeleteBulkEventArgs, ValueTask>? MessageDeleteBulk;
+
+    /// <summary>
+    /// Sent when a user adds a reaction to a message.
+    /// </summary>
     public event Func<MessageReactionAddEventArgs, ValueTask>? MessageReactionAdd;
+
+    /// <summary>
+    /// Sent when a user removes a reaction from a message.
+    /// </summary>
     public event Func<MessageReactionRemoveEventArgs, ValueTask>? MessageReactionRemove;
+
+    /// <summary>
+    /// Sent when a user explicitly removes all reactions from a message.
+    /// </summary>
     public event Func<MessageReactionRemoveAllEventArgs, ValueTask>? MessageReactionRemoveAll;
+
+    /// <summary>
+    /// Sent when a bot removes all instances of a given emoji from the reactions of a message.
```

Comment:
Is it only for a bot?

Reply:
Apparently? I know users _can_ do it, but discord doesn't seem to agree for some reason, I'll test it manually.

Reply:
Unsurprisingly, discord's docs are wrong.